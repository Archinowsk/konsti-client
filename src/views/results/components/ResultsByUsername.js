/* @flow */
import React from 'react'
import { useTranslation } from 'react-i18next'
import _ from 'lodash'
import type { StatelessFunctionalComponent, Element } from 'react'
import type { Result } from 'flow/result.flow'

type Props = {|
  results: $ReadOnlyArray<Result>,
|}

export const ResultsByUsername: StatelessFunctionalComponent<Props> = (
  props: Props
): Element<'div'> => {
  const { results } = props
  const { t } = useTranslation()

  const sortedResults = _.sortBy(results, [
    result => result.username.toLowerCase(),
  ])

  const resultsTable = sortedResults.map(result => (
    <div className='flex-table-column' key={result.username}>
      <div className='flex-table-row'>{result.username}</div>
      <div className='flex-table-row'>
        {result.enteredGame.gameDetails.title}
      </div>
      <div className='flex-table-row'>
        {result.enteredGame.gameDetails.location}
      </div>
    </div>
  ))

  const resultsByUsername = (
    <div className='flex-table-container'>
      <div className='flex-table-column flex-table-header'>
        <div className='flex-table-row'>{t('player')}</div>
        <div className='flex-table-row'>{t('gameTitle')}</div>
        <div className='flex-table-row'>{t('gameInfo.location')}</div>
      </div>
      {resultsTable}
    </div>
  )

  return resultsByUsername
}
