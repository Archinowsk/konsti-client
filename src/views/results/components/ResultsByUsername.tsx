import React, { FunctionComponent, ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import _ from 'lodash';

import { Result } from 'typings/result.typings';

export interface Props {
  results: readonly Result[];
}

export const ResultsByUsername: FunctionComponent<Props> = (
  props: Props
): ReactElement<'div'> => {
  const { results } = props;
  const { t } = useTranslation();

  const sortedResults = _.sortBy(results, [
    result => result.username.toLowerCase(),
  ]);

  const resultsTable = sortedResults.map(result => (
    <div className='flex-table-column' key={result.username}>
      <div className='flex-table-row'>{result.username}</div>
      <div className='flex-table-row'>
        {result.enteredGame.gameDetails.title}
      </div>
      <div className='flex-table-row'>
        {result.enteredGame.gameDetails.location}
      </div>
    </div>
  ));

  const resultsByUsername = (
    <div className='flex-table-container'>
      <div className='flex-table-column flex-table-header'>
        <div className='flex-table-row'>{t('player')}</div>
        <div className='flex-table-row'>{t('gameTitle')}</div>
        <div className='flex-table-row'>{t('gameInfo.location')}</div>
      </div>
      {resultsTable}
    </div>
  );

  return resultsByUsername;
};
