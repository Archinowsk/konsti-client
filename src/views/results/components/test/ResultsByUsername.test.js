// @flow
import React from 'react'
import { shallow } from 'enzyme'
import { ResultsByUsername } from '../ResultsByUsername'
import type { Props } from '../ResultsByUsername'

const results = []

describe('ResultsByUsername', () => {
  it('should render correctly', () => {
    const props: Props = { results }
    const component = shallow(<ResultsByUsername {...props} />)
    expect(component).toMatchSnapshot()
  })
})
