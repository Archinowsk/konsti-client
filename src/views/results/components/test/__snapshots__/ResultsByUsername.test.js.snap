// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsByUsername should render correctly 1`] = `
<div
  className="flex-table-container"
>
  <div
    className="flex-table-column flex-table-header"
  >
    <div
      className="flex-table-row"
    >
      player
    </div>
    <div
      className="flex-table-row"
    >
      gameTitle
    </div>
    <div
      className="flex-table-row"
    >
      gameInfo.location
    </div>
  </div>
</div>
`;
